flask_restful/__init__.py

```diff
@@ -272,15 +272,21 @@ def handle_error(self, e):
         """
         got_request_exception.send(current_app._get_current_object(), exception=e)
 
-        if not hasattr(e, 'code') and current_app.propagate_exceptions:
+        if not isinstance(e, HTTPException) and current_app.propagate_exceptions:
             exc_type, exc_value, tb = sys.exc_info()
             if exc_value is e:
                 raise
             else:
                 raise e
 
-        code = getattr(e, 'code', 500)
-        data = getattr(e, 'data', error_data(code))
+        if isinstance(e, HTTPException):
+            code = e.code
+            data = getattr(e, 'data', error_data(code))
```

Comment:
That test does seem to set up a somewhat unreasonable scenario.

Those failing tests should also probably raise the right exception instead of changing the `code
`test_handle_error` use `BadRequest`
`test_handle_auth` use `Unauthorized`
`test_handle_server_error` use `InternalServerError`